### PR TITLE
Use more `into_iter` rather than `drain(..)`

### PR DESCRIPTION
**Pull request by ldm0**
Clearer semantic.

### CODE REVIEW

File: compiler/rustc_data_structures/src/sorted_map.rs [hunk 1]
```diff
@@ -175,19 +175,19 @@ impl<K: Ord, V> SortedMap<K, V> {
 
         let drain = match start_index {
             Ok(index) => {
-                let mut drain = elements.drain(..);
+                let mut drain = elements.into_iter();
```
Comment (the8472) [inline compiler/rustc_data_structures/src/sorted_map.rs:178]: the variable name isn't appropriate anymore after the change

File: compiler/rustc_errors/src/translation.rs [hunk 2]
```diff
@@ -21,7 +21,7 @@ pub trait Translate {
     /// Typically performed once for each diagnostic at the start of `emit_diagnostic` and then
     /// passed around as a reference thereafter.
     fn to_fluent_args<'arg>(&self, args: &[DiagnosticArg<'arg>]) -> FluentArgs<'arg> {
-        FromIterator::from_iter(args.to_vec().drain(..))
+        FromIterator::from_iter(args.to_vec().into_iter())
```
Comment (the8472) [inline compiler/rustc_errors/src/translation.rs:24]: To create an intermediate Vec is redundant. `args.iter().cloned()` or `.copied()` should do the job too without extra allocation

File: compiler/rustc_data_structures/src/sorted_map.rs (hunk 1 above)
Comment (ldm0) [reply compiler/rustc_data_structures/src/sorted_map.rs]: Done

File: compiler/rustc_errors/src/translation.rs (hunk 2 above)
Comment (ldm0) [reply compiler/rustc_errors/src/translation.rs]: Nice

Comment (ldm0) [reply compiler/rustc_errors/src/translation.rs]: Done

**Pull request merged**